.github/workflows/ccruntime-on-push.yaml

```diff
@@ -0,0 +1,17 @@
+name: ccruntime e2e test on-push
+on:
+  pull_request_target:
```

Comment:
Changing from `pull_request` to `pull_request_target` will change the behavior of ` actions/checkout@v4`, i.e., it will check out the pull request tree which is not rebased with main. We will need to rebase the check'ed code ourselves to emulate the same behavior of `pull_request`. This is a well-known problem, we can fix it as kata CI and in a follow up PR if you prefer.

Reply:
Yeah, then I will borrow the script and place it in this PR rather than instituting it in the next PR.